### PR TITLE
adding trace compilation

### PR DESCRIPTION
**Pull request by mmogilnitsky**
trace is not published.

### CODE REVIEW

File: modules/canvaskit/.gitignore [hunk 1]
```diff
@@ -5,5 +5,7 @@ coverage/*
 # Don't check in this copy. It's just for publishing to npm.
 npm_build/CHANGELOG.md
 canvaskit-debug
-npm_build-debug/
+npm_build-debug*/
+npm_build-trace*/
```
Comment (ronpfeifle) [inline modules/canvaskit/.gitignore:9]: are the `*` characters on the correct side of the slash in these 2 lines?

File: modules/canvaskit/Makefile [hunk 2]
```diff
@@ -62,11 +62,38 @@ bluescape_release:
...
+	sed -i -e 's/\-bluescape\-/\-bluescape\-trace\-/g' ./npm_build-trace/package.json
+	cp ../../out/canvaskit_wasm/canvaskit.js   ./npm_build-trace/bin
+	cp ../../out/canvaskit_wasm/canvaskit.wasm ./npm_build-trace/bin
+
 bluescape_debug:
 	# Does an incremental build where possible.
 	# Note that we shouldn't disable: no_fonts no_paragraph primitive_shaper
-	rm -rf ../../out/canvaskit_wasm_debug
-	./compile.sh debug \
+	# rm -rf ../../out/canvaskit_wasm_debug
```
Comment (ronpfeifle) [inline modules/canvaskit/Makefile]: was this '#' comment intentional?

File: modules/canvaskit/Makefile [hunk 3]
```diff
@@ -62,11 +62,38 @@ bluescape_release:
...
+	./compile.sh \
+				debug \
+				force_tracing \
```
Comment (ronpfeifle) [inline modules/canvaskit/Makefile:96]: is force_tracing correct for debug?

File: modules/canvaskit/compile.sh [hunk 4]
```diff
@@ -56,6 +49,18 @@ else
   BUILD_DIR=${BUILD_DIR:="out/canvaskit_wasm"}
 fi
 
+# Tracing will be disabled unless this flag is seen.
+if [[ $@ != *force_tracing* ]] ; then
```
Comment (ronpfeifle) [inline modules/canvaskit/compile.sh]: I would reverse the 'if' and 'else' clauses so the test is == instead

**Review by ronpfeifle — COMMENTED** [review]
probably ok?  I had some nits (of course 😄)

File: modules/canvaskit/.gitignore (hunk 1 above)
Comment (mmogilnitsky) [reply modules/canvaskit/.gitignore:9]: yes.
I tried 
npm_build-debug2

File: modules/canvaskit/Makefile (hunk 2 above)
Comment (mmogilnitsky) [reply modules/canvaskit/Makefile]: thanks

File: modules/canvaskit/Makefile (hunk 3 above)
Comment (mmogilnitsky) [reply modules/canvaskit/Makefile:96]: yes it was expected to be so and it actually works

File: modules/canvaskit/compile.sh (hunk 4 above)
Comment (mmogilnitsky) [reply modules/canvaskit/compile.sh]: Just copied from previous but let me see.

**Review by ja2nicholl — APPROVED** [review]
Seems fine, I'm curious what happens if you have call the trace functions with trace disabled in a release build - are they no-ops?
Also, do you need to explicitly bump the version?

**Pull request merged**